admin/pkg/oauth/utils.go

```diff
@@ -7,8 +7,92 @@ const (
 
 // TokenResponse contains the information returned after fetching an access token from the OAuth server.
 type TokenResponse struct {
-	AccessToken string `json:"access_token"`
-	ExpiresIn   int64  `json:"expires_in,string"`
-	TokenType   string `json:"token_type"`
-	UserID      string `json:"user_id"`
+	AccessToken  string `json:"access_token"`
+	ExpiresIn    int64  `json:"expires_in"`
+	RefreshToken string `json:"refresh_token,omitempty"`
+	TokenType    string `json:"token_type"`
+	Scope        string `json:"scope,omitempty"`
+	UserID       string `json:"user_id"`
+}
+
+// ProtectedResourceMetadata contains the OAuth 2.0 Protected Resource Metadata as per RFC 8414
+// See: https://www.rfc-editor.org/rfc/rfc8414.html
+type ProtectedResourceMetadata struct {
```

Comment:
Did you check if one of the OAuth libraries we use already have these types from the RFC? Just to check if we need to declare them ourselves or can get them from a lib.